contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -4001,6 +4001,84 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql IMMUTABLE STRICT PARALLEL SAFE;
 
+-- wrapper functions for concat --
+CREATE OR REPLACE FUNCTION sys.concat(VARIADIC args sys.VARCHAR[] DEFAULT '{}')
+RETURNS sys.VARCHAR
+AS $$
+DECLARE
+    arr_len INTEGER;
+BEGIN
+    arr_len := array_length(args, 1);
+
+    -- Defined as per TSQL definition but PG has limitation for max number of args = 100
+    IF arr_len IS NULL OR arr_len < 2 OR arr_len > 254 THEN
```

Comment:
So if I supply more than 100 args to sys.concat, it will work as we are using ARRAY_TO_STRING function instead of concat of PG, right?

Reply:
The error is thrown by PG itself even before it reaches here, so it will not work, but amended the error message to allow a maximum of 100 arguments to be consistent with PG.